Source/EventFlow.EntityFramework/EventFlow.EntityFramework.csproj

```diff
@@ -22,7 +22,10 @@
   </PropertyGroup>
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="Microsoft.EntityFrameworkCore">
-      <Version>2.1.0</Version>
+      <Version>2.1.1</Version>
+    </PackageReference>
+    <PackageReference Include="linq2db.EntityFrameworkCore">
```

Comment:
My mainly concern was its performance. And we already have an alternative choice for testing (EF with in-memory SQLite).  But, we should not limit this library support to relational only, important pointed. I could revert to yours. I am not familiar with non relational databases, is there any EF core libraries with non relational, or any support plans? Appliciate your great review, thanks.

Reply:
I wasn't happy with my solution either, due to performance. EF Core has two main libraries: the core library and the relational library. If we depend on the relational library, our code works only in the context of relational databases. I don't know if this is a problem.

[In their EF Core docs](https://docs.microsoft.com/en-us/ef/core/providers/#future-providers) Microsoft states that they are building a Cosmos DB Provider which would be the first non-relational one (well, besides the InMemory implementation).

An alternative would be to create an interface for bulk deletion, put my hacky solution in there so that people can replace it with a better implementation that fits their specific database, and add a configuration method to make this easily configurable. This would be quite similar to how `UseUniqueConstraintDetectionStrategy` works, which also defaults to a hacky solution that won't work for everybody out of the box.

Reply:
Currently, the new interface idea looks good to me. I will try it. And, the default implementation would be yours. Yes, non-English-speaking world users like me (technically, who use .NET language pack) need to a specific implementation for `UseUniqueConstraintDetectionStrategy`.

Reply:
@frankebersoll There was no adding special interfaces for deletion method. Instead of this, we could decorate each store/persistence interface, or replace them completely via DI setting.
Also as a result of consideration the above, moved `BulkDeletionBatchSize` property to the new `IBulkOperationConfiguration` interface, same as `IUniqueConstraintDetectionStrategy`. (No needs this setting, if decorate or replace to other implementation.)